src/app.jsx

```diff
@@ -60,6 +63,18 @@ export class Application extends React.Component {
         this.setState({ alerts: alerts });
     }
 
+    checkExpiration(cert) {
+        const eventdate = moment(Number(cert["not-valid-after"].v) * 1000);
+        const todaysdate = moment();
+        const diffDays = eventdate.diff(todaysdate, "days");
+        const diffSeconds = eventdate.diff(todaysdate, "days");
+
+        if (diffSeconds < 0)
+            this.setState({ expiredCerts: this.state.expiredCerts++ });
+        else if (diffDays > 28)
+            this.setState({ toExpireCerts: this.state.toExpireCerts++ });
```

Comment:
editing state based on current state is much better when done "atomically" through `setState` callback. (`setState((prevState) => {...})`)

Reply:
Fixed